src/Elm/Ast.hs

```diff
@@ -79,25 +81,35 @@ data ElmPrim
     | ElmInt                                -- ^ @Int@
     | ElmFloat                              -- ^ @Float@
     | ElmString                             -- ^ @String@
-    | ElmTime                               -- ^ @Posix@ in elm, @UTCTime@ in Haskell
-    | ElmValue                              -- ^ @Json.Encode.Value@ in elm, @Data.Aeson.Value@ in Haskell
-    | ElmMaybe !TypeRef                     -- ^ @Maybe T@
-    | ElmResult !TypeRef !TypeRef           -- ^ @Result A B@ in elm
     | ElmPair !TypeRef !TypeRef             -- ^ @(A, B)@ in elm
     | ElmTriple !TypeRef !TypeRef !TypeRef  -- ^ @(A, B, C)@ in elm
-    | ElmList !TypeRef                      -- ^ @List A@ in elm
-    | ElmNonEmptyPair !TypeRef              -- ^ @NonEmpty A@ represented by @(A, List A)@ in elm
     deriving (Show)
 
+-- | Buitin types defined by core or 3rd party libraries
+-- Included definitions:
+-- * @Maybe a@
+-- * @Result a b@
+-- * @List a@
+-- * @Time.Posix@
+-- * @Json.Encode.Value@
+data ElmBuiltin = ElmBuiltin
+    { builtinImplType    :: !Text
+    , builtinImplEncoder :: !Text
```

Comment:
As a user of this API It's not clear from the type what this Text is supposed to be.
Is it name of the function defined somewhere? (Where?)
Can I add my custom definitions in that place?
Is it supposed to be fully qualified?
Although it's clear from the usage examples, it might not be so clear to library users ..

Reply:
this should be documented, hover before user will be able to define own things free-rely we need to make it possible to define custom imports for generated modules. So it doesn't make much sense documenting this before that part redesign so whole process could be documented.